zulipterminal/ui_tools/messages.py

```diff
@@ -798,14 +808,59 @@ def update_message_author_status(self) -> bool:
 
         return author_is_present
 
+    @staticmethod
+    def transform_content_alert_words(content: str, alerted_list: List[str]) -> Any:
+        alert_regex_replacements = {
+            "&": "&amp;",
+            "<": "&lt;",
+            ">": "&gt;",
+            # Accept quotes with or without HTML escaping
+            '"': r"(?:\"|&quot;)",
+            "'": r"(?:\'|&#39;)",
+        }
+
+        before_punctuation = r"\s|^|>|[\\(\".,';\\[]"
+
+        after_punctuation = r"(?=\s|$|<|[\\)\"?!:.,';\]!])"
+
+        clean: str
+
+        def replace_callback(match: Match[Union[str, str, str]]) -> str:
+            before = match.group(1)
+            word = match.group(2)
+            after = match.group(3)
+            offset = match.start()
+            matched_content = match.string
+            pre_match = matched_content[:offset]
+            check_string = pre_match + match.group()
+            in_tag = check_string.rfind("<") > check_string.rfind(">")
+            if in_tag:
+                return f"{before}{word}{after}"
+            return f"{before}<span class='alert-word'>{word}</span>{after}"
+
+        for word in alerted_list:
+            clean = "".join(alert_regex_replacements.get(c, c) for c in word)
+            regex = f"({before_punctuation})({clean})({after_punctuation})"
+            regex1 = re.compile(regex, re.IGNORECASE)
+            content = re.sub(regex1, replace_callback, content)
```

Comment:
The method `re.sub` has a parameter called `flags` in which you could pass the `IGNORECASE` flag, instead of defining a new regex. Ex: `re.sub(.., flags=re.IGNORECASE)`